linera-core/src/client/mod.rs

```diff
@@ -1230,11 +1220,9 @@ where
 
     async fn handle_certificate(
         &self,
-        admin_chain: ChainId,
         highest_known_epoch: Epoch,
-        committees: &mut BTreeMap<Epoch, Committee>,
+        committees: &BTreeMap<Epoch, Committee>,
         incoming_certificate: Certificate,
```

Comment:
Why not pass it in by reference, and make the result type a simple enum?

Reply:
You're right. Maybe in the original approach (pre-merge to `main`) it made more sense but now it's not true. 

Pass by reference and simplified the enum.